code/scripting/api/libs/ui.cpp

```diff
@@ -542,6 +558,208 @@ ADE_FUNC(getCmdBriefing,
+	"The promotion stage, name and bitmap or NIL if not earned")
+{
+	char filename[80];
+	SCP_string displayname;
+
+	if (Player->stats.m_promotion_earned != -1) {
+		Promoted = Player->stats.m_promotion_earned;
+		debrief_choose_medal_variant(filename, Rank_medal_index, Promoted);
+		displayname = Ranks[Promoted].name;
+
+		return ade_set_args(L, "oss", l_DebriefStage.Set(debrief_stage_h(&Promotion_stage)), displayname, filename);
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(getEarnedBadge,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the earned badge stage, name, and bitmap",
+	"debriefing_stage, string, string",
+	"The badge stage, name and bitmap or NIL if not earned")
+{
+	char filename[80];
+	SCP_string displayname;
+
+	if (Player->stats.m_badge_earned.size()) {
+		debrief_choose_medal_variant(filename,
+			Player->stats.m_badge_earned.back(),
+			Player->stats.medal_counts[Player->stats.m_badge_earned.back()] - 1);
+		displayname = Ranks[Promoted].name;
+
+		return ade_set_args(L, "oss", l_DebriefStage.Set(debrief_stage_h(&Badge_stage)), displayname, filename);
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(getTraitor,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Get the traitor stage",
+	"debriefing_stage",
+	"The traitor stage or nil if the player is not traitor. Will also clear out any earned stats if true")
+{
+	if (Turned_traitor) {
+		Player->flags &= ~PLAYER_FLAGS_PROMOTED;
+		scoring_level_init(&Player->stats);
+		return ade_set_args(L, "o", l_DebriefStage.Set(debrief_stage_h(&Traitor_debriefing.stages[0])));
+	} else {
+		return ADE_RETURN_NIL;
+	}
+}
+
+ADE_FUNC(mustReplay,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Gets whether or not the player must replay the mission. Will also clear out any earned stats if true",
+	"boolean",
+	"true if must replay, false otherwise")
+{
+
+	if (Must_replay_mission) {
+		Player->flags &= ~PLAYER_FLAGS_PROMOTED;
+		scoring_level_init(&Player->stats);
+		return ade_set_args(L, "b", true);
+	} else {
+		return ade_set_args(L, "b", false);
+	}
+}
+
+ADE_FUNC(canSkip,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Gets whether or not the player has failed enough times to trigger a skip dialog",
+	"boolean",
+	"true if can skip, false otherwise")
+{
+
+	if (Player->failures_this_session >= PLAYER_MISSION_FAILURE_LIMIT) {
+		return ade_set_args(L, "b", true);
+	} else {
+		return ade_set_args(L, "b", false);
+	}
+}
+
+ADE_FUNC(replayMission,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Resets the mission outcome and begins the current mission again if in campaign",
+	"number",
+	"Returns true when completed")
+{
+	// This is used to skip some UI preloading in debrief init
+	API_Access = true;
+
+	debrief_close();
+
+	gameseq_post_event(GS_EVENT_START_GAME);
+
+	API_Access = false;
+
+	return ade_set_args(L, "b", true);
+}
+
+ADE_FUNC(acceptMission,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Accepts the mission outcome, saves the stats, and begins the next mission if in campaign",
```

Comment:
Is campaign mode relevant to this function?

Reply:
For this one yes. It calls debrief_accept() which has explicit checks for what to do in campaign mode. If not in campaign made then it just send you back to the mainhall.

Reply:
👍

Reply:
Ok